### PR TITLE
fix: failed to search items by rss feed url which has query parameters

### PR DESCRIPTION
**Pull request by sywhb**
* encode uri component of the rss feed url
* use descriminated unions to differetiate rss and newsletter subscription

### CODE REVIEW

File: packages/api/src/services/library_item.ts [hunk 1]
```diff
@@ -321,17 +321,16 @@ export const buildQueryString = (
...
             default: {
-              let sql = 'library_item.archived_at IS NULL'
               if (useFolders) {
                 const param = `folder_${parameters.length}`
                 const folderSql = escapeQueryWithParameters(
                   `library_item.folder = :${param}`,
                   { [param]: value }
                 )
-                sql = `(${sql} AND ${folderSql})`
+                return `(library_item.archived_at IS NULL AND ${folderSql})`
```
Comment (jacksonh) [inline packages/api/src/services/library_item.ts:330]: Did you want these changes in this PR too?

Comment (sywhb) [reply packages/api/src/services/library_item.ts:330]: Thanks! I will move these backend changes to a separate PR

File: packages/web/pages/settings/feeds/index.tsx [hunk 2]
```diff
@@ -266,7 +271,11 @@ export default function Rss(): JSX.Element {
                 </VStack>
               }
               onClick={() => {
-                router.push(`/home?q=in:inbox rss:"${subscription.url}"`)
+                router.push(
+                  `/home?q=in:inbox rss:"${encodeURIComponent(
```
Comment (jacksonh) [inline packages/web/pages/settings/feeds/index.tsx:275]: Actually this should be `search?q=`

**Review by jacksonh — APPROVED** [review]

**Comment by jacksonh** [comment]
Just one small change as we don't use home for queries anymore.

File: packages/web/pages/settings/feeds/index.tsx (hunk 2 above)
Comment (sywhb) [reply packages/web/pages/settings/feeds/index.tsx]: Thanks!

**Pull request merged**